src/pages/search/index.tsx

```diff
@@ -46,13 +48,40 @@ const SearchPage: NextPage = () => {
   };
 
   const { data, isSuccess, isLoading, error } = useSearch(omit(['p'], params));
+  const toast = useToast();
+
+  const {
+    data: highlightsData,
+    isLoading: highlightsIsLoading,
+    isError: highlightsIsError,
+  } = useGetHighlights(omit(['p'], params), {
+    enabled: showHighlights,
+  });
+
+  useEffect(() => {
+    if (!highlightsIsLoading && highlightsIsError) {
+      toast({
+        status: 'error',
+        title: 'Error!',
+        description: 'Unable to fetch highlights',
+      });
+    }
+  }, [highlightsIsLoading, highlightsIsError]);
```

Comment:
I think we should move all this highlights related stuff into a hook at least or maybe build it into the toggle component if possible.  We can use the cache to pass the data around if needed.